src/resources/rmd/ojs_static.R

```diff
@@ -1,43 +1,47 @@
-attach(list(
-  ojs_define = function(...) {
-    # validate that we aren't in a cached chunk
-    cache <- knitr:::opts_current$get('cache')
-    if (is.numeric(cache) && cache > 0) {
-      stop("ojs_define() cannot be cached\n",
-           "Add cache: false to this cell's options to disable caching",
-           call. = FALSE)
-    }
-    quos <- rlang::enquos(...)
-    vars <- rlang::list2(...)
-    nm <- names(vars)
-    if (is.null(nm)) {
-      nm <- rep_len("", length(vars))
-    }
-    contents <- jsonlite::toJSON(list(contents = I(mapply(
+ojs_define <- function(...) {
+  # validate that we aren't in a cached chunk
+  cache <- knitr:::opts_current$get("cache")
+  if (is.numeric(cache) && cache > 0) {
+    stop("ojs_define() cannot be cached\n",
+      "Add cache: false to this cell's options to disable caching",
+      call. = FALSE
+    )
+  }
+  quos <- rlang::enquos(...)
+  vars <- rlang::list2(...)
+  nm <- names(vars)
+  if (is.null(nm)) {
+    nm <- rep_len("", length(vars))
+  }
+  contents <- jsonlite::toJSON(
+    list(contents = I(mapply(
       function(q, nm, val) {
         # Infer name, if possible
         if (nm == "") {
-          tryCatch({
-            nm <- rlang::as_name(q)
-          }, error = function(e) {
-            code <- paste(collapse = "\n", deparse(rlang::f_rhs(q)))
-            stop("ojs_define() could not create a name for the argument: ", code)
-          })
+          tryCatch(
+            {
+              nm <- rlang::as_name(q)
+            },
+            error = function(e) {
+              code <- paste(collapse = "\n", deparse(rlang::f_rhs(q)))
+              stop("ojs_define() could not create a name for the argument: ", code)
+            }
+          )
         }
-        list(name=nm, value=val)
-      }, quos, nm, vars, SIMPLIFY = FALSE, USE.NAMES=FALSE))),
-      dataframe = "columns", null = "null", na = "null", auto_unbox = TRUE)
-    script_string <- c(
-        "<script type=\"ojs-define\">",
-        contents,
-        "</script>");
-
-    # don't emit HTML output in PDF formats. (#2334)
-    if (knitr:::is_html_output(quarto_format$pandoc$to)) {
-      invisible(knitr:::knit_meta_add(list(structure(class="ojs-define", script_string))))
-    }
-  }
-), name = "tools:quarto")
-
-
-
+        list(name = nm, value = val)
+      }, quos, nm, vars,
+      SIMPLIFY = FALSE, USE.NAMES = FALSE
+    ))),
+    dataframe = "columns", null = "null", na = "null", auto_unbox = TRUE
+  )
+  script_string <- c(
+    "<script type=\"ojs-define\">",
+    contents,
+    "</script>"
+  )
+  invisible(
```

Comment:
Do we not need the `is_html_output()` check here anymore? Will this cause us to regress on #2334 ?

Reply:
No because I moved it with other dependency resolution where we include some for HTML and other for LaTeX / PDF

https://github.com/quarto-dev/quarto-cli/blob/255bcd1c58a44d9f7ba966133aeeccab63021745/src/resources/rmd/execute.R#L440-L450

So we include `ojs_define` in a special env only for HTML, like other HTML dependencies. (and not even for `prefer-html` as I figure we don't want it there (like hugo format))

Reply:
> I find it simpler to do all the conditional in one place, and this avoid adding format to a special env

I agree! I just missed that part. Thanks for clarifying.